numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1101,12 +1190,11 @@ TIMEDELTA_setitem(PyObject *op, void *ov, void *vap)
 /**begin repeat1
  *
  * #FROMTYPE = BYTE, UBYTE, SHORT, USHORT, INT, UINT, LONG, ULONG,
- *             LONGLONG, ULONGLONG, FLOAT, DOUBLE, LONGDOUBLE, DATETIME,
- *             TIMEDELTA#
+ *             LONGLONG, ULONGLONG, FLOAT, DOUBLE, LONGDOUBLE, DATETIME#
```

Comment:
So I've extracted timedelta from->to typecasting from the original template to start the specialized templates above. Otherwise I'd hit a ton of type conversions with an extraneous NaT check.